source/includes/limitations/4.3.rst

```diff
@@ -0,0 +1,113 @@
+Learn about the following TypeScript specific limitations of
+version {+version+} of the {+driver-short+}:
+
+- :ref:`The {+driver-short+} is unable to provide type safety for references to nested instances of recursive types specified through dot notation. <node-driver-recursive-types-dot-notation>`
+- :ref:`The {+driver-short+} does not support mutual recursion. <node-driver-limitations-mutual-recursion>`
+
+The TypeScript specific limitations of the {+driver-short+} relate to **recursive types**.
+A recursive type is a type that references itself. You can update
+the :ref:`Pet <mongodb-node-typescript-pet-interface>` interface
+to be recursive by allowing a pet to have its own pet. The following is the
+recursive ``Pet`` interface:
+
+.. _node-driver-limitations-recursive-pet:
+
+.. code-block:: typescript
+   :emphasize-lines: 2
+
+   interface RecursivePet {
+      pet?: RecursivePet;
+      name: string;
+      age: number;
+   }
```

Comment:
I believe mutual recursion is considered[ a form of recursion](https://en.wikipedia.org/wiki/Mutual_recursion#Data_types), and thus it can be said that all limitations relate to recursive types.

That said, I agree that moving this example into the recursive types section benefits the page, as readers do not need to understand the relationship between recursion and mutual recursion in order to avoid errors in their programs.

Reply:
I think there might be some confusion. The definition "recursive type is a type that references itself" does not cover mutual recursive types, which requires a second type. A mutually recursive type is one that needs to be recursive in terms of another type.